stmhal/mphalport.c

```diff
@@ -104,15 +104,15 @@ void mp_hal_gpio_clock_enable(GPIO_TypeDef *gpio) {
     } else if (gpio == GPIOH) {
         __GPIOH_CLK_ENABLE();
     #endif
-    #ifdef __GPIOI_CLK_ENABLE
+    #if defined(GPIOI) && defined(__GPIOI_CLK_ENABLE)
```

Comment:
`__GPIOI_CLK_ENABLE` on L4 is defined in `h1l/l4/inc/Legacy/stm32_hal_legacy.h`, line 2214 as `__HAL_RCC_GPIOI_CLK_ENABLE` and this is no further defined on the l4 platform. Therefore I added the test for GPIOI which is not defined on l4 platform and both functions are needed to fullfill the code. According to the Reference Manual (RM0351, P277) the port can be A.H and there is no I, J .... 

An alternative would be to check for the `MCU_SERIES_L4` define.